src/map/instance.cpp

```diff
@@ -708,6 +708,25 @@ int instance_addmap(int instance_id) {
 	return idata->map.size();
 }
 
+/**
+ * Fills outname with the name of the instance map name
+ * @param map_id: Mapid to use
+ * @param instance_id: Instance id
+ * @param outname: Pointer to allocated memory that will be filled in
+ */
+void instance_generate_mapname(int map_id, int instance_id, char outname[MAP_NAME_LENGTH]) {
+
+#if MAX_MAP_PER_SERVER > 9999
+	#error This algorithm is only safe for up to 9999 maps, change at your own risk.
+#endif
+	// Since we can't truncate integers like we can strings, allocate enough of a buffer
+	// to hold two integers, a symbol, and \0, then copy it to outname
+	char name[30];
+	snprintf(name, 30, "%d#%d", map_id, (instance_id % 1000000));
+	memcpy(outname, name, MAP_NAME_LENGTH - 1);
+	outname[MAP_NAME_LENGTH - 1] = 0;
```

Comment:
```suggestion
	// Safe up to 9999 maps per map-server
	const size_t prefix_length = 4;
	// Full map name length - prefix length - seperator character - zero termination
	const size_t suffix_length = MAP_NAME_LENGTH - prefix_length - 1 - 1;
	const int prefix_limit = (int)pow( 10, prefix_length );
	const int suffix_limit = (int)pow( 10, suffix_length );

	safesnprintf( outname, MAP_NAME_LENGTH, "%0*d#%0*d", prefix_length, map_id % prefix_limit, suffix_length, instance_id % suffix_limit );
```

Reply:
You can't truncate integers with printf, only strings. 

From http://www.cplusplus.com/reference/cstdio/printf/
> A format specifier follows this prototype: [see compatibility note below]
%[flags][width][.precision][length]specifier 

width | description
-- | --
(number) | Minimum number of characters to be  printed. If the value to be printed is shorter than this number, the  result is padded with blank spaces. The value is not truncated even if  the result is larger.


Because of this, we need to either convert the integers to strings, then truncate, or print the map name to a buffer, that's large enough to satisfy the compiler, then copy the result (which we know is of the correct size) to the destination buffer.